pkg/azure/provider/helpers/driver.go

```diff
@@ -36,12 +36,18 @@ import (
 	"github.com/gardener/machine-controller-manager-provider-azure/pkg/azure/utils"
 )
 
+// DataDiskLun describes the dataDisk lun
```

Comment:
Also there is no real need to define these types. Here is my argument. There are only 2 places that you are using the map[DataDiskLun]DiskID:
`driver.CreateDisksWithImageRef` where the usage is:
```go
disks[DataDiskLun(specDataDisk.Lun)] = disk.ID
```
and in `driver.getDataDisks` where the usage is:
```go
diskID := imageRefDiskIDs[DataDiskLun(specDataDisk.Lun)]
```

In both these usages there is an unnecessary cast. Also if you read these usages even without these type alias it is quite clear what is the key and what is the value of the map.
The only reason that i can think of to have a type alias is for method signatures but then you can just name the field appropriately so that it reflects what is the map all about.